File: Examples/test-suite/ocaml/fvirtual_runme.ml
open Swig
open Fvirtual

let _ =
  let sw = new_NodeSwitch '() and n = new_Node '() in
  assert (n -> addChild (n) as int = 1);
  assert (sw -> addChild (n) as int = 2);
  assert (sw -> addChild (sw) as int = 2);
  assert (sw -> addChild (n, false) as int = 3)
;;

File: Examples/test-suite/lua/fvirtual_runme.lua
require("import")
require("fvirtual")
f=fvirtual
catch_undef_globs() -- catch "undefined" global variables

n1 = f.Node()
n2 = f.Node()
assert(n1:addChild(n2) == 1)

ns = f.NodeSwitch()
assert(ns:addChild(n2) == 2)
assert(ns:addChild(ns) == 2)
assert(ns:addChild(n1, false) == 3)
